raster-foundry/raster-foundry
# Log error and return 400 for missing histograms when serving tiles

## Overview

Brief description of what this PR does, and why it is needed.

### Checklist


### Demo
```
curl -v 'http://localhost:8081/631a99a0-c43f-4c3f-80ca-0d5df7359b5a/9/235/216/?tag=1566488691251&token=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik9URXpORGRDUkRRM1JUa3hOVGc1TVVFME9UQTNOekJGTVRZeU0wTkRNekV5TXpoRk1qUXpOQSJ9.eyJodHRwczovL2FwcC5yYXN0ZXJmb3VuZHJ5LmNvbTtwbGF0Zm9ybSI6IjMxMjc3NjI2LTk2OGItNGU0MC04NDBiLTU1OWQ5YzY3ODYzYyIsImh0dHBzOi8vYXBwLnJhc3RlcmZvdW5kcnkuY29tO29yZ2FuaXphdGlvbiI6ImRmYWM2MzA3LWI1ZWYtNDNmNy1iZWRhLWI5ZjIwOGJiNzcyNiIsImh0dHBzOi8vYXBwLnJhc3RlcmZvdW5kcnkuY29tO2Fubm90YXRlQXBwIjp0cnVlLCJuaWNrbmFtZSI6ImRldiIsIm5hbWUiOiJkZXZAcmFzdGVyZm91bmRyeS5jb20iLCJwaWN0dXJlIjoiaHR0cHM6Ly9zLmdyYXZhdGFyLmNvbS9hdmF0YXIvMjMzNzI2YzQxYmI1NmYwYzY1YjA3NzlmNDQ5YmQ3NDY_cz00ODAmcj1wZyZkPWh0dHBzJTNBJTJGJTJGY2RuLmF1dGgwLmNvbSUyRmF2YXRhcnMlMkZkZS5wbmciLCJ1cGRhdGVkX2F0IjoiMjAxOS0wOC0yMlQxMzo0MTo1Ni40NjZaIiwiZW1haWwiOiJkZXZAcmFzdGVyZm91bmRyeS5jb20iLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiaXNzIjoiaHR0cHM6Ly9yYXN0ZXItZm91bmRyeS1kZXYuYXV0aDAuY29tLyIsInN1YiI6ImF1dGgwfDU5MzE4YTlkMmZiYmNhM2UxNmJjZmM5MiIsImF1ZCI6IjU0NEN6MVY4dHFlek1DYnlOOG1OSm1sek9wZDhINGV4IiwiaWF0IjoxNTY2NDgxMzE2LCJleHAiOjE1NjY1Njc3MTYsImF0X2hhc2giOiJ5OUtIU3NwWDFVTUozSVR1MmVyWURRIiwibm9uY2UiOiJNaWdQYXk1WWwwaGJvTVktTjA2Wmhxb19peUlsOWptYiJ9.DLLiYcEin6Vw68DJBAWE9XZRxC37PclUkRFg_OTgksrLrCRfInU9Qug4Ek6Dz17BpvAOzRAKjf4kNZ7WzeKUbaSR_wcCG8_YeaY1u307FD0JS5sLJjXhu8Rbxj-IGiCqqxTqp58F0PeNEYYPmUGkcTKBGY8qgQDGSEEUyL-rRj-HJtNyrB_249KO4wkm25c8uYiH6_2T6T1UKr5m9P33uKtaeCK3yN15j0W0p3ZTENzmbRRynoBlcxn85xmV4yH-JSXG7Zcv_ZgtSQus9JU4K2_dsa_VU4ybhIZulWSJrNeFb7516gO5YcBFn9wiWjKxVRPmQJbiNG6oZqXu5bNonA' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:69.0) Gecko/20100101 Firefox/69.0' -H 'Accept: image/webp,*/*' -H 'Accept-Language: en-US,en;q=0.5' --compressed -H 'DNT: 1' -H 'Connection: keep-alive' -H 'Referer: http://localhost:9091/projects/edit/631a99a0-c43f-4c3f-80ca-0d5df7359b5a/colormode' -H 'Cookie: intercom-session-w247m8f0=OG1kUE04MU5iTzI1OUtXcjl3QXpFQ3NIcW9zWUtYWTJFRDVPTHF6QU5nd2REeVY5Wmx5d1I2MU84SHArYlJPNy0tUE53OHdHbjY1d2VQVEhucVlNNmRSUT09--05268555fc4c593431d8a96866615be5d413e496; com.auth0.auth.v9xmxwVkj8pPDvbJGP7NLiM9iFAYaC4O={%22nonce%22:%22MigPay5Yl0hboMY-N06Zhqo_iyIl9jmb%22%2C%22state%22:%22v9xmxwVkj8pPDvbJGP7NLiM9iFAYaC4O%22%2C%22lastUsedConnection%22:%22Username-Password-Authentication%22}'
*   Trying ::1:8081...
* TCP_NODELAY set
* Connected to localhost (::1) port 8081 (#0)
> GET /631a99a0-c43f-4c3f-80ca-0d5df7359b5a/9/235/216/?tag=1566488691251&token=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik9URXpORGRDUkRRM1JUa3hOVGc1TVVFME9UQTNOekJGTVRZeU0wTkRNekV5TXpoRk1qUXpOQSJ9.eyJodHRwczovL2FwcC5yYXN0ZXJmb3VuZHJ5LmNvbTtwbGF0Zm9ybSI6IjMxMjc3NjI2LTk2OGItNGU0MC04NDBiLTU1OWQ5YzY3ODYzYyIsImh0dHBzOi8vYXBwLnJhc3RlcmZvdW5kcnkuY29tO29yZ2FuaXphdGlvbiI6ImRmYWM2MzA3LWI1ZWYtNDNmNy1iZWRhLWI5ZjIwOGJiNzcyNiIsImh0dHBzOi8vYXBwLnJhc3RlcmZvdW5kcnkuY29tO2Fubm90YXRlQXBwIjp0cnVlLCJuaWNrbmFtZSI6ImRldiIsIm5hbWUiOiJkZXZAcmFzdGVyZm91bmRyeS5jb20iLCJwaWN0dXJlIjoiaHR0cHM6Ly9zLmdyYXZhdGFyLmNvbS9hdmF0YXIvMjMzNzI2YzQxYmI1NmYwYzY1YjA3NzlmNDQ5YmQ3NDY_cz00ODAmcj1wZyZkPWh0dHBzJTNBJTJGJTJGY2RuLmF1dGgwLmNvbSUyRmF2YXRhcnMlMkZkZS5wbmciLCJ1cGRhdGVkX2F0IjoiMjAxOS0wOC0yMlQxMzo0MTo1Ni40NjZaIiwiZW1haWwiOiJkZXZAcmFzdGVyZm91bmRyeS5jb20iLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiaXNzIjoiaHR0cHM6Ly9yYXN0ZXItZm91bmRyeS1kZXYuYXV0aDAuY29tLyIsInN1YiI6ImF1dGgwfDU5MzE4YTlkMmZiYmNhM2UxNmJjZmM5MiIsImF1ZCI6IjU0NEN6MVY4dHFlek1DYnlOOG1OSm1sek9wZDhINGV4IiwiaWF0IjoxNTY2NDgxMzE2LCJleHAiOjE1NjY1Njc3MTYsImF0X2hhc2giOiJ5OUtIU3NwWDFVTUozSVR1MmVyWURRIiwibm9uY2UiOiJNaWdQYXk1WWwwaGJvTVktTjA2Wmhxb19peUlsOWptYiJ9.DLLiYcEin6Vw68DJBAWE9XZRxC37PclUkRFg_OTgksrLrCRfInU9Qug4Ek6Dz17BpvAOzRAKjf4kNZ7WzeKUbaSR_wcCG8_YeaY1u307FD0JS5sLJjXhu8Rbxj-IGiCqqxTqp58F0PeNEYYPmUGkcTKBGY8qgQDGSEEUyL-rRj-HJtNyrB_249KO4wkm25c8uYiH6_2T6T1UKr5m9P33uKtaeCK3yN15j0W0p3ZTENzmbRRynoBlcxn85xmV4yH-JSXG7Zcv_ZgtSQus9JU4K2_dsa_VU4ybhIZulWSJrNeFb7516gO5YcBFn9wiWjKxVRPmQJbiNG6oZqXu5bNonA HTTP/1.1
> Host: localhost:8081
> Accept-Encoding: deflate, gzip
> User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:69.0) Gecko/20100101 Firefox/69.0
> Accept: image/webp,*/*
> Accept-Language: en-US,en;q=0.5
> DNT: 1
> Connection: keep-alive
> Referer: http://localhost:9091/projects/edit/631a99a0-c43f-4c3f-80ca-0d5df7359b5a/colormode
> Cookie: intercom-session-w247m8f0=OG1kUE04MU5iTzI1OUtXcjl3QXpFQ3NIcW9zWUtYWTJFRDVPTHF6QU5nd2REeVY5Wmx5d1I2MU84SHArYlJPNy0tUE53OHdHbjY1d2VQVEhucVlNNmRSUT09--05268555fc4c593431d8a96866615be5d413e496; com.auth0.auth.v9xmxwVkj8pPDvbJGP7NLiM9iFAYaC4O={%22nonce%22:%22MigPay5Yl0hboMY-N06Zhqo_iyIl9jmb%22%2C%22state%22:%22v9xmxwVkj8pPDvbJGP7NLiM9iFAYaC4O%22%2C%22lastUsedConnection%22:%22Username-Password-Authentication%22}
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 400 Bad Request
< Server: nginx
< Date: Thu, 22 Aug 2019 16:29:30 GMT
< Content-Type: text/plain; charset=UTF-8
< Content-Length: 101
< Connection: keep-alive
< Strict-Transport-Security: max-age=15552000; preload
< X-Content-Type-Options: nosniff
< 
* Connection #0 to host localhost left intact
No histogram stored for scene in layer: 8fa3bb21-50eb-4e18-984d-5c052959d825. Please re-ingest scene.% 
```

![image](https://user-images.githubusercontent.com/4392704/63532508-d52d3300-c4d8-11e9-8ad8-4a41f03a3a30.png)

## Testing Instructions

- Choose a project and scene. Delete the histogram from the layer_attributes table.
- Change the color mode, and verify that tile requests for non-empty tiles return a 400 error and explanation.
- Verify that the error is logged.

Closes https://github.com/raster-foundry/raster-foundry/issues/5100
